sdv/data_processing/data_processor.py

```diff
@@ -556,6 +556,8 @@ def _fit_hyper_transformer(self, data):
         """
         if not data.empty:
             self._hyper_transformer.fit(data)
+        else:
```

Comment:
Can we move this up a level so that it doesn't call any of the constraint stuff?